PluginAutotask/API/Read/ReadRecordsTicketHistory.cs

```diff
@@ -102,9 +130,26 @@ public static async IAsyncEnumerable<Record> ReadRecordsTicketHistoryAsync(IApiC
                     }
 
                     var queryWrapper = JsonConvert.DeserializeObject<QueryWrapper>(await queryResult.Content.ReadAsStringAsync());
-                    foreach (var rawRecord in queryWrapper.Items) 
+                    if (isRangedTicketHistory)
```

Comment:
This block is too similar to the one on L69. Could have been reused by making a function.

Reply:
Agreed. That would look something like this:

```cs
var queryWrapper = JsonConvert.DeserializeObject<QueryWrapper>(await queryResult.Content.ReadAsStringAsync());
var records = PullTicketHistoryRecordsFromWrapper(schema, queryWrapper, queryDate);
foreach (var record in records) // or await foreach if we need ^^^ method to be an Async Enumerable
    yield return record;
```

Reply:
Although it doesn't change the plugin in a drastic way, I'd say we should try to include this change sooner instead of making a second pass by after testing. Sending this change up shortly.